core/validation.py

```
@@ -8,15 +8,30 @@
+from urllib.parse import urlsplit
 
 
 def validate_server_url(url: str) -> tuple[bool, str | None]:
     """Validate server URL format."""
     if not url or not url.strip():
         return False, "Server URL is empty"
     clean = url.strip()
-    if not re.match(r"^https?://[^/\s]+", clean):
+    try:
+        parts = urlsplit(clean)
+    except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)